src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -800,6 +801,53 @@ def get_local_connections_from_executable(executable):
     return result
 
 
+def resolve_annotation(anno) -> Union[str, list]:
+    """Resolve the union annotation to type list."""
+    origin = get_origin(anno)
+    if origin != Union:
+        return anno
+    # Optional[Type] is Union[Type, NoneType], filter NoneType out
+    args = [arg for arg in get_args(anno) if arg != type(None)]  # noqa: E721
+    return args[0] if len(args) == 1 else args
+
+
+def param_to_definition(param, value_type) -> (InputDefinition, bool):
+
+    def value_to_str(val):
+        if val is inspect.Parameter.empty:
```

Comment:
not sure if this method covered all supported values, double confirm and create all possible tests.